src/components/site-search/site-search.client.ts

```diff
@@ -92,8 +99,59 @@ requestAnimationFrame(() => {
 					}
 
 					searchResults.replaceChildren(hdom)
+					resultChildren = searchResults.querySelectorAll('.listitem')
+					resultChildren.forEach((result, index) => {
+						if (index === 0) {
+							result.setAttribute('aria-selected', 'true')
+						} else {
+							result.setAttribute('aria-selected', 'false')
+						}
+					})
 				})
 			})
 		})
+		let indexNumber = 0
+		const onKeydown = (event: { key: string; preventDefault: () => void }) => {
```

Comment:
If you don’t prefer this, or if it doesn’t work perfectly the first time, totally ignore it, and seriously, we should only blame TypeScript.

```suggestion
		const onKeydown = (event: KeyboardEvent) => {
```